modules/codegen/src/main/scala/com/twilio/guardrail/core/Tracker.scala

```diff
@@ -0,0 +1,126 @@
+    def toNel: Tracker[Option[Mappish[NonEmptyList, A, B]]] = tracker.map(x => NonEmptyList.fromList(x.value).map(Mappish(_)))
+  }
+}
+
+trait HighPriorityTrackerSyntax extends LowPriorityTrackerSyntax {
+  implicit class StringyEitherSyntax[B](tracker: Tracker[Either[String, B]]) {
+    def raiseErrorIfLeft: Target[Tracker[B]] = tracker.fold(err => Target.raiseError(s"${err.get} (${err.showHistory})"), Target.pure _)
+  }
+
+  implicit class EitherSyntax[A, B](tracker: Tracker[Either[A, B]]) {
+    def fold[C](a: Tracker[A] => C, b: Tracker[B] => C): C =
+      tracker.unwrapTracker.fold(x => a(Tracker.cloneHistory(tracker, x)), x => b(Tracker.cloneHistory(tracker, x)))
+  }
+
+  implicit class OptionSyntax[A](tracker: Tracker[Option[A]]) {
+    def orHistory: Either[Vector[String], Tracker[A]]      = tracker.indexedCosequence.toRight(tracker.history)
+    def raiseErrorIfEmpty(err: String): Target[Tracker[A]] = Target.fromOption(tracker.indexedCosequence, s"${err} (${tracker.showHistory})")
+    class FlatDownFieldPartiallyApplied[C](val dummy: Boolean = true) {
+      def apply[B](label: String, f: A => B)(implicit ev: Tracker.Convincer[Option[B], C]): Tracker[C] =
+        new Tracker(ev(tracker.get.flatMap(x => Option(f(x)))), tracker.history :+ s".${label}")
+    }
+    def flatDownField[C]: FlatDownFieldPartiallyApplied[C] = new FlatDownFieldPartiallyApplied()
+  }
+
+  implicit class RefineSyntax[A](tracker: Tracker[A]) {
+    class RefinePartiallyApplied[C](val dummy: Boolean = true) {
+      def apply[B1](r: PartialFunction[A, B1])(f: Tracker[B1] => C): Either[Tracker[A], C] =
+        r.andThen(value => f(tracker.map(_ => value)))
+          .andThen(Right(_))
+          .applyOrElse(tracker.get, (_: A) => Left(tracker))
+    }
+    def refine[C]: RefinePartiallyApplied[C] = new RefinePartiallyApplied[C]()
+  }
+
+  implicit class RefineEitherSyntax[A, C](value: Either[Tracker[A], C]) {
+    def orRefine[B1](r: PartialFunction[A, B1])(f: Tracker[B1] => C): Either[Tracker[A], C] =
+      value.fold({ tracker =>
+        r.andThen(value => f(tracker.map(_ => value)))
+          .andThen(Right(_))
+          .applyOrElse(tracker.get, (_: A) => Left(tracker))
+      }, Right(_))
+  }
+
+  implicit class Syntax[A](tracker: Tracker[A]) {
+    class DownFieldPartiallyApplied[C](val dummy: Boolean = true) {
+      def apply[B](label: String, f: A => B)(implicit ev: Tracker.Convincer[Option[B], C]): Tracker[C] =
+        new Tracker(ev(Option(f(tracker.get))), tracker.history :+ s".${label}")
+    }
+    def downField[C]: DownFieldPartiallyApplied[C] = new DownFieldPartiallyApplied()
+
+    def showHistory: String = tracker.history.mkString("")
+
+    @deprecated("Tracker.get will be removed once the migration has been completed. Please use the fold/traverse combinators to get values out.", "0.0.0")
+    def get: A = tracker.get
+
+    def unwrapTracker: A = tracker.get
```

Comment:
Just to test my understanding: `unwrapTracker` is for places where you really do need to just pull the value out and there's no possibility to fold or traverse, whereas `get` is for places where things aren't yet done the right way, and need to be revisited later?

Reply:
Exactly that. I added `get` initially, but then it was difficult to disambiguate between `Option.get` (which is regrettably present) and `tracker.get`. I marked `get` as deprecated and added this.